test/endpoints/baz/baz_metrics_test.go

```diff
@@ -146,7 +147,77 @@ func TestCallMetrics(t *testing.T) {
 		assert.Equal(t, expectedEndpoitTags[tag.GetTagName()], tag.GetTagValue())
 	}
 
-	tchannelOutboundSuccessMetric := metrics[3]
+	expectedJaegerSpanMetricName := "test-gateway.production.all-workers.jaeger.spans"
```

Comment:
We setup a metrics backend in `TestGateway` ( https://github.com/uber/zanzibar/blob/master/test/lib/test_gateway/test_gateway.go#L185 ). This is a good place to setup a jaeger reporter backend ( using `MockAgent` ).